### PR TITLE
dialects: Add missing Symbol traits.

### PR DESCRIPTION
**Pull request by PapyChacal**
Add all missing direct SymbolOpInterface on implemented operations.
Our implemetations were alright, so no changes needed on anything else than declaring the trait :grin: Will allow more generic handlign of those though!

NB: some operations are "transitively" Symbols, because FunctionOpInterface => Symbol, and we don't have FunctionOpInterface yet. I decided to leave those alone for now!

@mesham I'll wait for your feedback, as this makes a change on the FIR dialect, and I'm not sure if you rely on this in your PsyClone flow? This didn't break any test, but I want to be sure I don't break anything untested before merging this!

### CODE REVIEW

**Review by compor — APPROVED** [review]
I'm fine with this, but I only checked `llvm`, `memref`, `pdl` and `gpu`.

File: xdsl/dialects/experimental/fir.py [hunk 1]
```diff
@@ -1142,8 +1143,12 @@ class DispatchTable(IRDLOperation):
     """
 
     name = "fir.dispatch_table"
+
+    sym_name = attr_def(StringAttr)
```
Comment (math-fehr) [inline xdsl/dialects/experimental/fir.py:1147]: I guess that in the tests, `sym_name` was used but not required?

**Review by math-fehr — APPROVED** [review]
Nice! Thanks!

File: xdsl/dialects/experimental/fir.py (hunk 1 above)
Comment (PapyChacal) [reply xdsl/dialects/experimental/fir.py:1147]: I don't know, I'm not sure it's tested tbh! I chatted with @mesham and ensure this won't be an issue short term either way 🙂

**Pull request merged**